### PR TITLE
simplify FS put behavior for partial files

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by kannappanr** [comment]
PTAL at build failures

**Comment by harshavardhana** [comment]
> PTAL at build failures

#3247 should be taken first @kannappanr and then I shall rebase this PR.

File: cmd/client-fs.go [hunk 1]
```diff
@@ -316,59 +306,34 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
...
 	}
 
-	// Close the file before rename.
-	if e = partFile.Close(); e != nil {
-		return totalWritten, probe.NewError(e)
-	}
+	// Close the file before renaming, we need to do this
+	// specifically for windows users - windows explicitly
+	// disallows renames on Open() fd's by default.
+	tmpFile.Close()
```
Comment (vadmeste) [inline cmd/client-fs.go:336]: Let's return an error only in this tmpFile.Close() (easier to debug when one user complains)

Comment (harshavardhana) [reply cmd/client-fs.go:322]: @vadmeste why are we bothered about error from Close() in which scenario?

**Pull request merged**